server/server.js

```diff
@@ -63,17 +63,19 @@ async function handleValidRequest(button, numPresses, batteryLevel) {
     let session = await db.getUnrespondedSessionWithButtonId(button.button_id, client)
         
     if (session === null) {
-        session = await db.createSession(button.installation_id, button.button_id, button.unit, button.phone_number, numPresses, client)
+        session = await db.createSession(button.installation_id, button.button_id, button.unit, button.phone_number, numPresses, batteryLevel, client)
```

Comment:
Maybe this needs to be moved below the 
```
    if ((batteryLevel !== undefined) &&
            (batteryLevel >= 0) &&
            (batteryLevel <= 100)) {
```
condition so that you can find out whether the value of `batteryLevel` is even valid to put into the DB. If it's not, then you can save `null` in that column.

Reply:
oh! I hadn't accounted for the possibility that we could be saving -1 or some other non-null and also non-valid battery value, that's a good point